cql3/statements/permission_altering_statement.cc

```diff
@@ -45,6 +48,17 @@ future<> cql3::statements::permission_altering_statement::check_access(query_pro
     maybe_correct_resource(_resource, state, qp);
 
     return state.ensure_exists(_resource).then([this, &state] {
+        if (_resource.kind() == auth::resource_kind::functions) {
+            // Even if the function exists, it may be a builtin function, in which case we disallow altering permissions on it.
+            auth::functions_resource_view v(_resource);
+            if (v.function_signature()) {
+                auto [name, function_args] = auth::decode_signature(*v.function_signature());
+                auto fun = cql3::functions::functions::find(db::functions::function_name{sstring(*v.keyspace()), name}, function_args).get();
+                if (!dynamic_cast<cql3::functions::user_function*>(fun) && !dynamic_cast<cql3::functions::user_aggregate*>(fun)) {
```

Comment:
There is an is_native() method.

What happens if there is no signature?

Reply:
Please add a comment then.

Reply:
Ok